packages/blade/src/components/NativeAria/NativeAria.native.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Pressable, Text, View } from 'react-native';
+import mapA11yProps from '../../utils/mapProps';
+import { CloseIcon, EyeIcon } from '../Icons';
+
+const CheckboxItem: React.FC<{ label: string; checked?: boolean; disabled?: boolean }> = ({
+  label,
+  checked,
+  disabled,
+}) => {
+  const [isChecked, setChecked] = React.useState(checked);
+
+  const toggleChecked = (): void => {
+    if (disabled) return;
+    setChecked((prev) => !prev);
+  };
+
+  return (
+    <>
+      {isChecked ? (
+        <EyeIcon color="action.icon.link.active" size="small" />
+      ) : (
+        <CloseIcon color="action.icon.link.active" size="small" />
+      )}
+      <Pressable
+        onPress={toggleChecked}
+        {...mapA11yProps({
+          accessibilityDisabled: disabled,
+          accessibilityRole: 'checkbox',
+          accessibilityChecked: isChecked ? 'true' : 'false',
```

Comment:
I think what you are suggesting is something similar to accessibility hooks. 

where we can do `useCheckboxA11y` and this hook will return the necessary accessibility props. 

But even with that we will need some sort of compatibility layer because that useCheckboxA11y hook needs to return proper props based on platform. so in a way this getProps function can be used internally within the useCheckboxA11y.

Reply:
Also mapA11yProps function is a general low level utility, not a component specific function. 

Usage as follows

```
mapA11yProps({ accessibiltyRole: 'button', accessibilityLabel: "Search" })

// web: -> { role: 'button', 'aria-label': 'search' } 
// native: -> { accessibiltyRole: 'button', accessibilityLabel: "Search" }
```

Reply:
bumping this so we can discuss this